### PR TITLE
Osd metadata command

### PR DESCRIPTION
**Pull request by mzhong1**
Added a command for osd metadata, as well as fixing the mgr metadata command (ceph mgr metadata and ceph osd metadata return json Maps).

### CODE REVIEW

File: src/cmd.rs [hunk 1]
```diff
@@ -55,11 +55,105 @@ pub struct MgrMetadata {
     pub hostname: String,
     pub kernel_description: String,
     pub kernel_version: String,
-    pub mem_swap_kb: u64,
-    pub mem_total_kb: u64,
+    pub mem_swap_kb: String,
```
Comment (cholcombe973) [inline src/cmd.rs]: I seem to recall ceph changing this field type somewhere along the way.  What version did you test against for this?

**Review by cholcombe973 — COMMENTED** [review]
Thanks for doing this!  I'm curious what version of ceph did you test against when creating this code?  Ceph unfortunately has changed the field types of the json output over the years.

File: src/cmd.rs (hunk 1 above)
Comment (mzhong1) [reply src/cmd.rs]: Luminous

Comment (mzhong1) [reply src/cmd.rs]: Actually, I could probably make this an enum, so yanno, either it gets serialized as u64 or String.  hm.

Comment (mzhong1) [reply src/cmd.rs]: I should probably also make those struct attributes public too

Comment (mzhong1) [reply src/cmd.rs]: Eh actually wait that's implied

File: src/cmd.rs [hunk 2]
```diff
@@ -1052,8 +1166,8 @@ pub fn mgr_disable_module(cluster_handle: &Rados, module: &str, simulate: bool)
     Ok(())
 }
 
-/// dump metadata for all daemons
-pub fn mgr_metadata(cluster_handle: &Rados) -> RadosResult<MgrMetadata> {
+/// dump metadata for all daemons.  Note this only works for Luminous+
+pub fn mgr_metadata(cluster_handle: &Rados) -> RadosResult<Vec<MgrMetadata>> {
```
Comment (cholcombe973) [inline src/cmd.rs:1171]: if this is luminous and newer i think the macro should be added here:
`min_version!(Luminous, self);`

**Comment by cholcombe973** [comment]
I think this looks fine.  We're going to need to do a major release version because this changes some public function return values.

File: src/cmd.rs (hunk 2 above)
Comment (mzhong1) [reply src/cmd.rs:1171]: Isn't that macro for ceph_client though?

**Pull request closed without merging**